sagemaker-pyspark-sdk/tests/namepolicy_test.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+import os
+
+from sagemaker_pyspark import (CustomNamePolicyFactory, classpath_jars,
+                               CustomNamePolicyWithTimeStampSuffixFactory,
+                               CustomNamePolicyWithTimeStampSuffix,
+                               CustomNamePolicy)
+
+from pyspark import SparkConf, SparkContext
+from py4j.java_gateway import JavaObject
+
+
+@pytest.fixture(autouse=True)
+def with_spark_context():
+    os.environ['SPARK_CLASSPATH'] = ":".join(classpath_jars())
+    conf = (SparkConf()
+            .set("spark.driver.extraClassPath", os.environ['SPARK_CLASSPATH']))
+
+    if SparkContext._active_spark_context is None:
+        SparkContext(conf=conf)
+
+    yield SparkContext._active_spark_context
+
+    # TearDown
+    SparkContext.stop(SparkContext._active_spark_context)
+
+
+def test_CustomNamePolicyFactory():
+    java_obj = CustomNamePolicyFactory("jobName", "modelname", "epconfig", "ep")
+    assert(isinstance(java_obj._to_java(), JavaObject))
+
+
+def test_CustomNamePolicyWithTimeStampSuffixFactory():
+    java_obj = CustomNamePolicyWithTimeStampSuffixFactory("jobName", "modelname", "epconfig", "ep")
```

Comment:
technically java_obj is really a python object. (applies to all the tests).

```python
name_policy = CustomName...()
# this is the actual java object
java_obj = name_policy._to_java()
```

Also, again this test is not really accomplishing anything. I would rather have a test that checks that the TrainingJobName in the java_object is set correctly.  You can accomplish this by doing

```python
name_policy._call_java("trainingJobName") == "jobName"
```

Applies to all the similar tests.